vql/server/splunk.go

```diff
@@ -204,15 +207,43 @@ func send_to_splunk(
 	var events []*splunk.Event
 
 	for _, event := range buf {
-		events = append(
-			events,
-			client.NewEvent(
-				event,
-				arg.Source,
-				arg.Sourcetype,
-				arg.Index,
-			),
-		)
+		if arg.TimestampField != "" {
+			dict, ok := event.(*ordereddict.Dict)
```

Comment:
You need to materialize the row here because the field may be lazy. https://cs.github.com/Velocidex/vfilter/blob/7af3cd80b93459c49cc346fa1894f7c55e3eefe7/lazy.go#L171 it will also ensure it is an ordered dict.

Similarly you can use TimeFromAny to ensure it is a timestamp https://cs.github.com/Velocidex/velociraptor/blob/0b1c605a8803917be61fd2c4a9477160edaaded8/vql/functions/time.go#L197

What happens if a timestamp is missing? should we fill in the current time, reject the row or completely give up?

Reply:
If some timestamps are null, I think they should probably be set to the beginning of the Epoch rather than current time. If there's no timestamp specified then I guess current time is fine.